File: videoBase/src/main/java/com/scccy/videoBase/config/manager/XBogusManager.java
package com.scccy.videoBase.config.manager;

import com.scccy.videoBase.untils.XbogusUtil;

import java.util.HashMap;
import java.util.Map;

public class XBogusManager {

    public static String str_2_endpoint(String user_agent, String endpoint) {
        try {

            String final_endpoint = XbogusUtil.getXBogus(endpoint);
            return String.valueOf(final_endpoint.charAt(0)); // Assuming XB.getXBogus() returns a String
        } catch (Exception e) {
            e.printStackTrace(); // Logging or handling the exception as needed
            throw new RuntimeException("生成X-Bogus失败: " + e.getMessage());
        }
    }

    public static String model_2_endpoint(String user_agent, String base_endpoint, Map<String, String> params) {
        if (params == null || !(params instanceof HashMap)) {
            throw new IllegalArgumentException("参数必须是HashMap类型");
        }

        String param_str = String.join("&", params.entrySet().stream()
                .map(entry -> entry.getKey() + "=" + entry.getValue())
                .toArray(String[]::new));

        try {
            String xb_value = XbogusUtil.getXBogus(param_str);
            String separator = base_endpoint.contains("?") ? "&" : "?";
            String final_endpoint = base_endpoint + separator + param_str + "&X-Bogus=" + xb_value.charAt(1); // Assuming xb_value is a String
            return final_endpoint;
        } catch (Exception e) {
            e.printStackTrace(); // Logging or handling the exception as needed
            throw new RuntimeException("生成X-Bogus失败: " + e.getMessage());
        }
    }
}

File: common/common-core/src/main/java/XBogusManager.py
class XBogusManager:
    @classmethod
    def str_2_endpoint(
        cls,
        user_agent: str,
        endpoint: str,
    ) -> str:
        try:
            final_endpoint = XB(user_agent).getXBogus(endpoint)
        except Exception as e:
            logger.error(traceback.format_exc())
            raise RuntimeError(_("生成X-Bogus失败: {0})").format(e))

        return final_endpoint[0]

    @classmethod
    def model_2_endpoint(
        cls,
        user_agent: str,
        base_endpoint: str,
        params: dict,
    ) -> str:
        if not isinstance(params, dict):
            raise TypeError(_("参数必须是字典类型"))

        param_str = "&".join([f"{k}={v}" for k, v in params.items()])

        try:
            xb_value = XB(user_agent).getXBogus(param_str)
        except Exception as e:
            logger.error(traceback.format_exc())
            raise RuntimeError(_("生成X-Bogus失败: {0})").format(e))

        # 检查base_endpoint是否已有查询参数 (Check if base_endpoint already has query parameters)
        separator = "&" if "?" in base_endpoint else "?"

        final_endpoint = f"{base_endpoint}{separator}{param_str}&X-Bogus={xb_value[1]}"

        return final_endpoint